### PR TITLE
Add Certificate Revocation Verification configurations

### PR DESCRIPTION
**Pull request by Bhashinee**
## Purpose
Add annotations and struct fields to server and client connector to enable certificate revocation.

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal [hunk 1]
```diff
@@ -14,6 +14,9 @@ package ballerina.net.http;
 @Field {value:"sslEnabledProtocols: SSL/TLS protocols to be enabled"}
 @Field {value:"ciphers: List of ciphers to be used"}
 @Field {value:"sslProtocol: The SSL protocol version"}
+@Field {value:"certificateRevocationVerifier: true/false to enable/disable certificate revocation verifier"}
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Shall we align the description with others? Eg The status of certificate revocation verifier {default value : true (enable)}

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java [hunk 2]
```diff
@@ -892,7 +896,18 @@ private static void extractHttpsConfig(Annotation configInfo, Set<ListenerConfig
...
             }
-
+            if (certificateRevocationVerifierAttrValue.getBooleanValue() && cacheDelayAttrValue == null) {
+                throw new BallerinaException("cacheDelay value must be provided to enable certificate validation");
+            }
+            if (certificateRevocationVerifierAttrValue.getBooleanValue() && cacheSizeAttrValue == null) {
+                throw new BallerinaException("cacheSize value must be provided to enable certificate validation");
+            }
+            if (certificateRevocationVerifierAttrValue.getBooleanValue()) {
+                listenerConfiguration
```
Comment (chamil321) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/HttpUtil.java]: Shall we move above two validations inside this if condition? Then we can get rid of redundant certificateRevocationVer...AttrValue.getBooleanValue().

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal [hunk 3]
```diff
@@ -37,6 +40,9 @@ public annotation configuration attach service<> {
     string sslEnabledProtocols;
     string ciphers;
     string sslProtocol;
+    boolean certificateRevocationVerifier;
+    int cacheSize;
+    int cacheDelay;
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/annotation.bal]: Is this a cache specific to certificate handling? If so, can we rename these fields to something which reflects that?

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 4]
```diff
@@ -279,6 +282,9 @@ public struct SSL {
     string sslEnabledProtocols;
     string ciphers;
     string sslProtocol;
+	boolean certificateRevocationVerifier;
+	int cacheSize;
+	int cacheDelay;
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: Same comment as for the `annotation.bal`

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java [hunk 5]
```diff
@@ -180,6 +183,9 @@
     public static final int SSL_ENABLED_PROTOCOLS_INDEX = 4;
     public static final int CIPHERS_INDEX = 5;
     public static final int SSL_PROTOCOL_INDEX = 6;
+    public static final int CERTIFICATE_REVOCATION_VERIFIER_INDEX = 0;
+    public static final int CACHE_SIZE = 0;
+    public static final int CACHE_DELAY = 1;
```
Comment (pubudu91) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/Constants.java]: These 2 are also indices right? If so, let's change this to conform to the format followed by other indices.

File: pom.xml [hunk 6]
```diff
@@ -1183,7 +1183,7 @@
         <equinox.osgi.version>3.10.2.v20150203-1939</equinox.osgi.version>
         <equinox.osgi.services.version>3.4.0.v20140312-2051</equinox.osgi.services.version>
         <carbon.kernel.version>5.1.0</carbon.kernel.version>
-        <transport.http.version>6.0.81</transport.http.version>
+        <transport.http.version>6.0.84</transport.http.version>
```
Comment (shafreenAnfar) [inline pom.xml]: This has to be the latest I suppose.

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 7]
```diff
@@ -285,6 +288,9 @@ public struct SSL {
     string sslEnabledProtocols;
     string ciphers;
     string sslProtocol;
+    boolean certificateRevocationVerifier;
```
Comment (shafreenAnfar) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: shall we change the name here as well. Similar to the one in the transport.

File: stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal [hunk 8]
```diff
@@ -285,6 +288,9 @@ public struct SSL {
...
+    int cacheSize;
+    int cacheDelay;
```
Comment (shafreenAnfar) [inline stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/natives.bal]: cacheValidityPeriod ?

File: stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java [hunk 9]
```diff
@@ -138,7 +138,20 @@ private void populateSenderConfigurationOptions(SenderConfiguration senderConfig
             String sslEnabledProtocols = ssl.getStringField(HttpConstants.SSL_ENABLED_PROTOCOLS_INDEX);
             String ciphers = ssl.getStringField(HttpConstants.CIPHERS_INDEX);
             String sslProtocol = ssl.getStringField(HttpConstants.SSL_PROTOCOL_INDEX);
+            boolean certificateRevocationVerifier =
+                    ssl.getBooleanField(HttpConstants.CERTIFICATE_REVOCATION_VERIFIER_INDEX) == TRUE;
+            int cacheSize = (int) ssl.getIntField(HttpConstants.CACHE_SIZE_INDEX);
+            int cacheDelay = (int) ssl.getIntField(HttpConstants.CACHE_DELAY_INDEX);
```
Comment (shafreenAnfar) [inline stdlib/ballerina-http/src/main/java/org/ballerinalang/net/http/actions/Init.java]: here also shall we change the name.

**Pull request merged**